packages/web-app-files/src/components/SideBar/Shares/InviteCollaborator/ExpirationDatepicker.vue

```diff
@@ -50,105 +50,112 @@ import { DateTime } from 'luxon'
 export default {
   name: 'ExpirationDatepicker',
   props: {
-    expirationDate: {
-      type: Date,
+    shareTypes: {
+      type: Array,
       required: false,
-      default: undefined
+      default: () => []
     }
   },
 
   data() {
     return {
-      enteredExpirationDate: null
+      expirationDateInternal: null
     }
   },
 
   computed: {
     ...mapGetters(['capabilities']),
 
-    expirationSupported() {
-      return this.userExpirationDate && this.groupExpirationDate
-    },
-
-    defaultExpirationDateSet() {
-      return this.userExpirationDate.enabled || this.groupExpirationDate.enabled
+    expirationDate: {
+      get: function () {
+        return this.expirationDateInternal || this.expirationDateDefault
+      },
+      set: function (val) {
+        this.expirationDateInternal = val
+
+        const expirationDate = DateTime.fromJSDate(val)
+          .setLocale(this.$language.current)
+          .endOf('day')
+        this.$emit('optionChange', {
+          expirationDate: expirationDate.isValid
+            ? expirationDate.toFormat("yyyy-MM-dd'T'HH:mm:ssZZZ")
+            : null
+        })
+      }
     },
 
-    userExpirationDate() {
-      return this.capabilities.files_sharing.user.expire_date
-    },
+    expirationDateDefault() {
+      if (!(this.expirationDateUser.enabled || this.expirationDateGroup.enabled)) {
+        return null
+      }
 
-    groupExpirationDate() {
-      return this.capabilities.files_sharing.group.expire_date
-    },
+      const hasUserCollaborators = this.shareTypes.includes(0)
+      const hasGroupCollaborators = this.shareTypes.includes(1)
+      const userMaxExpirationDays = parseInt(this.expirationDateUser.days, 10)
+      const groupMaxExpirationDays = parseInt(this.expirationDateGroup.days, 10)
 
-    defaultExpirationDate() {
-      if (!this.defaultExpirationDateSet) {
-        return null
+      let days = null
+      if (hasUserCollaborators) {
+        days = userMaxExpirationDays
       }
 
-      const userMaxExpirationDays = parseInt(this.userExpirationDate.days, 10)
-      const groupMaxExpirationDays = parseInt(this.groupExpirationDate.days, 10)
-      let days = 0
+      if (hasGroupCollaborators) {
+        days = groupMaxExpirationDays
+      }
 
-      if (userMaxExpirationDays && groupMaxExpirationDays) {
+      if (hasUserCollaborators && hasGroupCollaborators) {
         days = Math.min(userMaxExpirationDays, groupMaxExpirationDays)
-      } else {
-        days = userMaxExpirationDays || groupMaxExpirationDays
+      }
+
+      if (!days) {
+        return null
       }
 
       return DateTime.now().setLocale(this.$language.current).plus({ days }).toJSDate()
     },
 
+    expirationSupported() {
```

Comment:
Shouldn't it be something like `expirationDefaultExists` or something similar?

Reply:
i changed the naming now completly  and stopped exposing non public used values